// Copyright 2015 Google Inc. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
// License for the specific language governing permissions and limitations under
// the License.

syntax = "proto2";


package dsrf;

enum DataType {
  STRING = 0;
  INTEGER = 1;
  DECIMAL = 2;
  BOOLEAN = 3;
}

message Cell {

  // The name of the cell in the spreadsheet (eg. "RecordType").
  optional string name = 1;

  // The type of value the cell contains. This value will indicate which one of
  // the below fields will be populated.
  optional DataType cell_type = 2;

  // Contains string values if the cell is of type STRING.
  repeated string string_value = 3;

  // Contains integer values if the cell is of type INTEGER.
  repeated int64 integer_value = 4;

  // Contains decimal values if the cell is of type DECIMAL.
  repeated double decimal_value = 5;

  // Contains the boolean values if the cell is of type BOOLEAN.
  repeated bool boolean_value = 6;

}
